### DISCUSSION TITLE
Shadow DOM

### DISCUSSION THREAD

**Discussion by peacepostman** [nanojsx/nano#122] [Help / Questions]
Hello !

I'm discovering the nanoJSX library and I was wondering how does the library live within a shadow DOM ?

Best regards,
Ben

**Comment by yandeu** [comment]
I have never used a shadow DOM and am therefore not much familiar with its concept.
Shadow DOM is the DOM inside a custom HTMLElement?

In the [docs, at the bottom](https://nanojsx.io/docs.html) is an example of creating a custom element using NanoJSX.

This feature was requested and implement by @jrson83.

**Reply by peacepostman** [reply]
Yes it is the DOM inside a custom HTMLElement but you could also choose to inject your component in a Shadow DOM target. I'm using this lib with: react https://github.com/Wildhoney/ReactShadow

Regarding the behaviour of nano I think it is totally doable. I will perform some tests on my side to be sure.

Thanks !

**Comment by felippe-regazio** [comment]
Maybe you would like to see the "CustomElementsMode" in the docs: [CustomElementsMode](https://nanojsx.io/docs.html).
When you register a NanoJSX Component as a web element, it will be rendered inside a ShadowDOM by default.
